### PR TITLE
Adds recipe for Thrax

### PR DESCRIPTION
**Pull request by kylebgorman**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by kylebgorman** [comment]
Dear @conda-forge/help-c-cpp please review at your leisure.

File: recipes/thrax/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+./configure --prefix="${PREFIX}"
+make -j"${CPU_COUNT}"
+make -j"${CPU_COUNT}" install
```
Comment (scopatz) [inline recipes/thrax/build.sh]: Is there a `make check`?

File: recipes/thrax/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make
+  host:
+    - openfst ==1.7.5
+  run:
+    - openfst ==1.7.5
```
Comment (scopatz) [inline recipes/thrax/meta.yaml]: Same, does this need such a tight pinning?

File: recipes/thrax/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    - python
+
+test:
+  requires:
+    - openfst ==1.7.5
```
Comment (scopatz) [inline recipes/thrax/meta.yaml]: you don't need to duplicate deps that are also in the run-deps here.  This section is for test-only deps

File: recipes/thrax/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  commands:
+    - thraxcompiler --helpshort >/dev/null || [[ $? == 1 ]]
```
Comment (scopatz) [inline recipes/thrax/meta.yaml]: Same questions about redirection and exit codes

File: recipes/thrax/build.sh (hunk 1 above)
Comment (kylebgorman) [reply recipes/thrax/build.sh]: Yes, but it is the same as a bare `make`.

File: recipes/thrax/meta.yaml (hunk 2 above)
Comment (kylebgorman) [reply recipes/thrax/meta.yaml]: In this case, yes. (I am an upstream maintainer for both OpenFst and Thrax.) Basically every release of OpenFst involves massive API/ABI breaking, if only by nature of it being mostly a template library. (Upstream also has what are probably thousands of test units but they're proprietary to Google.)

File: recipes/thrax/meta.yaml (hunk 3 above)
Comment (kylebgorman) [reply recipes/thrax/meta.yaml]: Fixed, thanks.

File: recipes/thrax/meta.yaml (hunk 4 above)
Comment (kylebgorman) [reply recipes/thrax/meta.yaml]: Good point, fixed.

**Comment by scopatz** [comment]
Looks like this has conflicts now somehow.  I am guessing you used the same dir name accidentally

**Comment by kylebgorman** [comment]
Something else bad happened when I tried to pull from remote to merge the conflicts (the staged `envs` package is giving me trouble). I'm going to just start this over since I can't figure out how to fix it. @scopatz I'll add you back in.

**Pull request closed without merging**